portfolio/src/main/webapp/script.js

```diff
@@ -241,3 +241,9 @@ function addLocation(map, lat, lng, collegeName, internName) {
     info.open(map, marker); 
   });
 }
+
+function checkUserLogin() {
+  fetch('/login-user').then(response => response.json()).then(userStatus => {
```

Comment:
Did you mean to return the result of this fetch (so that checkUserLogin() returns Promise<bool> containing login status?

Reply:
Yes, so do I need to have another function that I pass the promise boolean to and return it from there?

Reply:
Sorry I didn't see the comment here until now.

What you have here works. You could also just return userStatus.loggedIn as you were doing before, then chain displayForm whereever you need it like:

```js
getUserStatus().then(displayForm)
```

(in this context, displayForm is a `function` accepting a boolean and returning nothing, which is the exact type `then()` takes in this context)

Reply:
This only works if you `return fetch` though.